### PR TITLE
Bug 1673991 - Set attribution when qs item is clicked

### PR DESCRIPTION
**Pull request by daleharvey**

### CODE REVIEW

**Comment by daleharvey** [comment]
(copied across since last PR was based on confusing branch)

@mikedeboer You mentioned using the exising module for attribution, should we be using the module as is or do we need to change the existing functionality / give it a custom url.

Also will we want different releases of this extension to opt in / out of attribution? if so I expect we should make it part of the data configuration.

File: scripts/build-addon.js [hunk 1]
```diff
@@ -30,7 +31,16 @@ let inputOptions = {
         { src: "src/*.js", dest: OUTDIR },
         { src: "src/icons/*", dest: `${OUTDIR}/icons` },
       ]
-    })
+    }),
+    {
+        name: "watch-external",
```
Comment (daleharvey) [inline scripts/build-addon.js:36]: Previously rollup watch would only reload the extension if the files used in the rollup build were changed (KeywordTree*.js) this makes sure it reloads if background.js / api.js etc change

**Comment by mikedeboer** [comment]
LGTM, but I think @Standard8 should formally review here 😉 

Have to say I really like the new load point for the dataset. I do wonder if it needs to be at an idleTimeout instead, though...

File: src/background.js [hunk 2]
```diff
@@ -56,10 +57,6 @@ class ProviderDynamicQuickSuggest extends UrlbarProvider {
 
   cancelQuery(queryContext) {}
 
-  pickResult(result) {
-    console.log("Result picked!", result);
-  }
```
Comment (Standard8) [inline src/background.js:61]: We probably just want to make this a dummy function, or remove the call from the shim.

File: src/experiments/urlbar/api.js [hunk 3]
```diff
@@ -34,10 +35,10 @@ async function time(fun) {
 }
 
 this.experiments_urlbar = class extends ExtensionAPI {
+  onStartup() {
+    time(() => treeProvider.load(this.extension.rootURI));
```
Comment (Standard8) [inline src/experiments/urlbar/api.js]: As Mike says, I think we want to `Services.tm.dispatchToMainThread` this, or do it on idle (though if we do idle, then we'll have to ensure it does happen if we haven't got to idle).

File: src/experiments/urlbar/api.js [hunk 4]
```diff
@@ -13,6 +13,7 @@ const { XPCOMUtils } = ChromeUtils.import(
 );
 
 XPCOMUtils.defineLazyModuleGetters(this, {
+  PartnerLinkAttribution: "resource:///modules/PartnerLinkAttribution.jsm",
```
Comment (Standard8) [inline src/experiments/urlbar/api.js]: I don't think we actually need to load this now.

**Review by Standard8 — APPROVED** [review]

**Comment by daleharvey** [comment]
Cheers, updated based on comments

**Pull request merged**